web/app.fluidity.money/app/routes/$network/dashboard/airdrop/common.tsx

```diff
@@ -468,14 +469,66 @@ const StakeNowModal = ({
     });
   };
 
-  const canStake = () => false;
+  const [canStake, setCanStake] = useState(false);
+
+  useEffect(() => {
+    (async () => {
+      setCanStake(await testStake());
+    })();
+  }, [baseToken, fluidToken, slippage]);
+
+  const testStake = async (): Promise<boolean> => {
+    try {
+      await testStakeTokens?.(
+        new BN(daysToSeconds(stakingDuration)),
+        baseToken.symbol === "USDC"
+          ? snapToValidValue(baseToken.amount, baseToken)
+          : new BN(0),
```

Comment:
Using BN and BigNumber in the same Ethereum-centric codebase when only one should be used?